lib/segment/src/vector_storage/simple_multi_dense_vector_storage.rs

```diff
@@ -64,10 +75,25 @@ pub fn open_simple_multi_dense_vector_storage(
             deleted_count += 1;
         }
         let point_id_usize = point_id as usize;
-        if point_id_usize >= vectors.len() {
-            vectors.resize(point_id_usize + 1, TypedMultiDenseVector::placeholder(dim));
+        if point_id_usize >= vectors_metadata.len() {
+            vectors_metadata.resize(point_id_usize + 1, Default::default());
         }
-        vectors[point_id_usize] = stored_record.vector;
+
+        let metadata = &mut vectors_metadata[point_id_usize];
+        metadata.size = stored_record.vector.flattened_vectors.len();
+        metadata.capacity = metadata.size;
```

Comment:
Why is the capacity equals to size here?
I thought it was the capacity left in the vector :thinking:

Reply:
`capacity` here is a size of allocated place in `ChunkedVectors`. When we rewrite vector, we can use old allocated place if new size is not larger than capacity

Reply:
Here it's equal because we allocate exact `size` vectors in `ChunkedVectors` container. When user rewrites vector with smaller multivector, size will be changed but capacity won't